### PR TITLE
feat: add option to transform text in filter options

### PR DESCRIPTION
**Pull request by zlatanpham**
Support options for transforming text for `list`, `select` and `tab` filter

### CODE REVIEW

**Comment by zlatanpham** [comment]
@tuanddd just hold up reviewing if you are going to do it. I think I need to add the functionality for other types of `Filter` as well.

File: packages/search-ui/src/Filter/TabFilter.tsx [hunk 1]
```diff
@@ -44,15 +45,15 @@ const TabFilter = (props: Omit<TabFilterProps, 'type'>) => {
           className={customClassNames.filter?.tabs?.tab}
           selectedClassName={customClassNames.filter?.tabs?.selectedTab}
         >
-          {t('all')}
+          {formatLabel('all', { textTransform, t })}
```
Comment (tuanddd) [inline packages/search-ui/src/Filter/TabFilter.tsx:48]: Screen Shot 2021-04-15 at 12 59 12
Screen Shot 2021-04-15 at 13 03 10
Difference between [new change](https://deploy-preview-461--sajari-react-sdk-v3.netlify.app/search-ui/filter#tabs) and [production](https://react.docs.sajari.com/search-ui/filter#tabs)

I think this should be
```suggestion
          {formatLabel(t('all'), { textTransform })}
```

**Review by tuanddd — CHANGES_REQUESTED** [review]
I thought we agreed that the default text transform is capitalize?

File: packages/search-ui/src/Filter/TabFilter.tsx (hunk 1 above)
Comment (zlatanpham) [reply packages/search-ui/src/Filter/TabFilter.tsx]: Fixed but we still need to pass `t` in.

**Comment by zlatanpham** [comment]
> I thought we agreed that the default text transform is capitalize?

we can do it in the `search-widget`.

**Review by tuanddd — APPROVED** [review]

**Pull request merged**